### PR TITLE
Update E0033 to the new error format

### PR DESCRIPTION
**Pull request by munyari**
Part of #35233

Addresses #35498
"r? @jonathandturner

### CODE REVIEW

**Comment by jonathandturner** [comment]
Looks good.

@bors r+ rollup

**Comment by jonathandturner** [comment]
@bors r-

Looks like the test file may need another label. This might come from other PRs that have landed. 

Line 16 on the unit test needs one more NOTE:

unexpected errors (from JSON output): [
    Error {
        line_num: 16,
        kind: Some(
            Note
        ),
        msg: "the trait `SomeTrait` cannot be made into an object"
    }
]

**Comment by munyari** [comment]
@jonathandturner looks like it comes from my other PR actually, because it doesn't show up in the unit tests of this branch rebased on master. I added the note

**Comment by jonathandturner** [comment]
Thanks.  Let's see if that did it.

@bors r+ rollup

File: src/test/compile-fail/E0033.rs [hunk 1]
```diff
@@ -15,5 +15,10 @@ trait SomeTrait {
 fn main() {
     let trait_obj: &SomeTrait = SomeTrait; //~ ERROR E0425
                                            //~^ ERROR E0038
-    let &invalid = trait_obj; //~ ERROR E0033
+                                           //~| method `foo` has no receiver
+                                           //~| NOTE the trait `SomeTrait` cannot be made into an object
```
Comment (eddyb) [inline src/test/compile-fail/E0033.rs]: Please move these to the left, indented *under* the erroring line, otherwise they're too wide (over 100 columns).

**Comment by munyari** [comment]
@eddyb just updated

File: src/test/compile-fail/E0033.rs [hunk 2]
```diff
@@ -13,7 +13,13 @@ trait SomeTrait {
 }
 
 fn main() {
-    let trait_obj: &SomeTrait = SomeTrait; //~ ERROR E0425
-                                           //~^ ERROR E0038
-    let &invalid = trait_obj; //~ ERROR E0033
+    let trait_obj: &SomeTrait = SomeTrait;
+    //~ ERROR E0425
```
Comment (eddyb) [inline src/test/compile-fail/E0033.rs]: The `~^` needs to be on this line.

**Comment by munyari** [comment]
@eddyb just amended the previous commit

File: src/test/compile-fail/E0033.rs [hunk 3]
```diff
@@ -13,7 +13,13 @@ trait SomeTrait {
 }
 
 fn main() {
-    let trait_obj: &SomeTrait = SomeTrait; //~ ERROR E0425
-                                           //~^ ERROR E0038
-    let &invalid = trait_obj; //~ ERROR E0033
+    let trait_obj: &SomeTrait = SomeTrait;
+    //~^ ERROR E0425
+    //~| ERROR E0038
+    //~| method `foo` has no receiver
```
Comment (eddyb) [inline src/test/compile-fail/E0033.rs]: Does this actually work? I didn't think it worked without `ERROR` or `NOTE`.

**Comment by jonathandturner** [comment]
@munyari - can you update the unit test with @eddyb's suggestion?  After that, I think we're good.

**Comment by munyari** [comment]
@jonathandturner @eddyb the tests seem to work on my end, and looks like it passed on Travis. Please tell me about any errors that you see

**Comment by eddyb** [comment]
Well, I could find other tests using the same syntax, so it may just be something new (although I would've expected @jonathandturner to explain in that case. I *believe* it's the new error label).

@bors r=jonathandturner

**Pull request merged**